trieste/acquisition/function/entropy.py

```diff
@@ -228,10 +231,13 @@ class SupportsCovarianceObservationNoiseTrajectory(
     has an associated trajectory sampler."""
 
 
-GIBBONT = TypeVar("GIBBONT", bound=SupportsCovarianceObservationNoise, contravariant=True)
+GIBBONModelType = TypeVar(
+    "GIBBONModelType", bound=SupportsCovarianceObservationNoise, contravariant=True
+)
+""" Type variable bound to :class:`~trieste.models.SupportsCovarianceObservationNoise`. """
```

Comment:
Why do we not have one of these in the cts Thompson sampling acq? That needs the model to have a sampler

Reply:
In general we don't need these just to specify that you need a specific feature: e.g. GreedyContinuousThompsonSampling is defined as a SingleModelGreedyAcquisitionBuilder[HasTrajectorySampler]. You only need type variables if the features you require depend on one of the input arguments.

For example, a rule that gets passed an acqusition function builder that only supports models with predict_joint also only supports models with predict_joint. Or an acquisition function (such as GIBBON) that gets passed a sampler that only supports models with trajectory functions also only supports models with trajectory functions.